_data-prepper/pipelines/expression-syntax.md

```diff
@@ -208,3 +218,30 @@ White space is **required** surrounding set initializers, priority expressions,
 | `==`, `!=`           | Equality operators       | No                   | `/status == 200`<br>`/status_code==200`                        |                                       |
 | `and`, `or`, `not`   | Conditional operators    | Yes                  | `/a<300 and /b>200`                                            | `/b<300and/b>200`                     |
 | `,`                  | Set value delimiter      | No                   | `/a in {200, 202}`<br>`/a in {200,202}`<br>`/a in {200 , 202}` | `/a in {200,}`                        |
+
+
+## Functions
+
+Data Prepper supports the following built-in functions that can be used in an expression.
+
+### `length()`
+
+The `length()` function takes one argument of the JSON pointer type and returns the length of the value passed. For example, `length(/message)` returns a length of `10`, when a key message exists in the event and has a value of `1234567890`.
+
+### `hasTags()`
+
+The `hastags()` function takes one or more string type arguments and returns `true` if all the arguments passed are present in an event's tags. When an argument does not exist in the event's tags, the function returns `false`. For example, if you use the express `hasTags("tag1")` and the event contains `tag1`, Data Prepper returns `true`. If you use the same expression but the event also contains  `tag4` tag, Data Prepper returns `false`.
```

Comment:
> If you use the same expression but the event **also** contains `tag4` tag, Data Prepper returns `false`.

Do you mean the event contains both `tag1` and `tag4`? 
If that's the case, `hasTags("tag1")` still returns `true`. `hasTags("tag4")` or `hasTags("tag1", "tag4")` also returns true. But `hasTags("tag2")` or `hasTags("tag1", "tag2")` would return false.